src/client_side.cc

```diff
@@ -2987,16 +2997,23 @@ ConnStateData::switchToHttps(ClientHttpRequest *http, Ssl::BumpMode bumpServerMo
     if (insideConnectTunnel)
         preservingClientData_ = shouldPreserveClientData();
 
-    readSomeData();
+    if (inBuf.isEmpty())
+        readSomeData();
+    else
+        parseTlsHandshake();
```

Comment:
@chtsanti, what do you think about this solution to avoid excessive InsufficientInput exceptions? See 33bfd69 for Eduard's thoughts on this problem.

Reply:
Yep looks good.
Sorry for late response.